pubsub-protocol-binding.md

```diff
@@ -0,0 +1,276 @@
+mandate specific existing features to be used.
+
+## 2. Use of CloudEvents Attributes
+
+This specification does not further define any of the [CloudEvents][ce] event
+attributes.
+
+This mapping is intentionally robust against changes, including the addition and
+removal of event attributes, and also accommodates vendor extensions to the
+event metadata.
+
+### 2.1. datacontenttype Attribute
+
+The `datacontenttype` attribute is assumed to contain a [RFC2046][rfc2046]
+compliant media-type expression.
+
+### 2.2. data Attribute
+
+The `data` attribute is assumed to contain opaque application data that is
+encoded as declared by the `datacontenttype` attribute.
+
+An application is free to hold the information in any in-memory representation
+of its choosing, but as the value is transposed into a Pub/Sub message as defined 
+in this specification, the assumption is that the `data` attribute value is made
+available as a sequence of bytes.
+
+For instance, if the declared `datacontenttype` is
+`application/json;charset=utf-8`, the expectation is that the `data` attribute
+value is made available as [UTF-8][rfc3629] encoded JSON text Pub/Sub message data.
+
+## 3. Pub/Sub Message Mapping
+
+With Pub/Sub, the content mode is chosen by the sender of the
+event. Protocol usage patterns that might allow solicitation of events using a
+particular content mode might be defined by an application, but are not defined
+here.
+
+The receiver of the event can distinguish between the two content modes by
+inspecting the [Pub/Sub Attribute `Content-Type`][content-type] of the
+[Pub/Sub message][pubsub-message]. If the attribute is present and its value is prefixed 
+with the CloudEvents media type `application/cloudevents`, indicating the use of a known
+[event format](#14-event-formats), the receiver uses _structured_ mode,
+otherwise it defaults to _binary_ mode.
+
+If a receiver finds a CloudEvents media type as per the above rule, but with an
+event format that it cannot handle, for instance
+`application/cloudevents+avro`, it MAY still treat the event as binary and
+forward it to another party as-is.
+
+### 3.1. Binary Content Mode
+
+The _binary_ content mode accommodates any shape of event data, and allows for
+efficient transfer and without transcoding effort.
+
+#### 3.1.1. Content-Type
+
+For the _binary_ mode, the [Pub/Sub Attribute `Content-Type`][content-type] of the 
+[Pub/Sub message][pubsub-message] property, if present, MUST be mapped
+directly to the CloudEvents `datacontenttype` attribute.
+
+#### 3.1.2. Event Data Encoding
+
+The [`data`](#22-data-attribute) byte-sequence MUST be used as the value of the Pub/Sub
+message data payload.
+
+#### 3.1.3. Metadata Attributes
+
+All [CloudEvents][ce] attributes 
+MUST be individually mapped to and from distinct Pub/Sub message attributes, with
+exceptions noted below.
+
+CloudEvents extensions that define their own attributes MAY define a diverging
```

Comment:
nit: "divergent". "Diverging" implies continuing change. Or just use "different" which is clearer and also apt.

Reply:
I think this was introduced at the time that extensions could be nested maps... So I ended up removing it... not entirely sure though. WDYT

Reply:
Now that extensions must have untyped string attribute values only, I can't think of a case where such a mapping would be required. I think it's fine to remove it. We can add this section back later if needed.

Reply:
sounds good!